pkg/kv/kvserver/kvserverbase/bulk_adder.go

```diff
@@ -67,6 +67,10 @@ type BulkAdderOptions struct {
 	// different from the timestamp used to construct the adder which is what is
 	// actually applied to each key).
 	BatchTimestamp hlc.Timestamp
+
+	// WriteAtRequestTime is used to set the corresponding field when sending
+	// constructed SSTables to AddSSTable. See roachpb.AddSSTableRequest.
+	WriteAtRequestTime bool
```

Comment:
You may want to tie this to the `BatchTimestamp` field above, which is the timestamp that will initially be used (unless the request then gets pushed).

It appears that only the index backfiller actually sets `BatchTimestamp`, to the `ReadAsOf` timestamp, but it doesn't set `WriteAtRequestTime` so that shouldn't be affected.

Reply:
we're going to get rid of BatchTimestamp with the new backfill anyway I think